src/main/java/edu/byu/cs/dataAccess/sql/SubmissionSqlDao.java

```diff
@@ -80,30 +79,43 @@ public Collection<Submission> getAllLatestSubmissions() throws DataAccessExcepti
 
     @Override
     public Collection<Submission> getAllLatestSubmissions(int batchSize) throws DataAccessException {
-        return sqlReader.executeQuery(
-                """
-                    WHERE timestamp IN (
-                        SELECT MAX(timestamp)
-                        FROM %s
-                        GROUP BY net_id, phase
-                    )
-                    ORDER BY timestamp DESC
-                    """.formatted(sqlReader.getTableName()) +
-                (batchSize >= 0 ? "LIMIT ?" : ""),
-                ps -> {
-                    if (batchSize >= 0) {
-                        ps.setInt(1, batchSize);
-                    }
-                });
+        try (var connection = SqlDb.getConnection()) {
+            var statement = connection.prepareStatement(
+                    """
+                            SELECT s.net_id, s.repo_url, s.timestamp, s.phase, s.passed, s.score, s.num_commits, s.head_hash, s.notes, s.rubric, s.admin
+                            FROM submission s
+                            INNER JOIN (
+                                SELECT net_id, phase, MAX(timestamp) AS max_timestamp
+                                FROM submission
+                                GROUP BY net_id, phase
+                            ) s2 ON s.net_id = s2.net_id AND s.phase = s2.phase AND s.timestamp = s2.max_timestamp
+                            ORDER BY s2.max_timestamp DESC
+                            """ +
```

Comment:
@frozenfrank 
I didn't foresee this issue when reviewing your big SQL PR, but I realized with this that your select helper doesn't cover queries that use `join`